src/routes/productRouter.ts

```diff
@@ -1,13 +1,46 @@
-import {Router} from "express"
+/* eslint-disable comma-dangle */
+import { Router } from "express";
 import productController from "../modules/product/controller/productController";
 import { userAuthorization } from "../middlewares/authorization";
-import { validation, isProductExist, isShopExist, transformFilesToBody } from "../middlewares/validation";
-import { shopSchema, productSchema} from "../modules/product/validation/productValidation";
+import {
+  validation,
+  isProductExist,
+  isShopExist,
+  transformFilesToBody,
+  isProductNotExist,
+} from "../middlewares/validation";
+import {
+  shopSchema,
+  productSchema,
+} from "../modules/product/validation/productValidation";
 import upload from "../helpers/multer";
 
-const router: Router = Router()
+const router: Router = Router();
 
-router.post("/create-product", userAuthorization(["seller"]), upload.array("images"), transformFilesToBody,validation(productSchema), isProductExist, productController.createProduct);
-router.post("/create-shop", userAuthorization(["seller"]), validation(shopSchema), isShopExist, productController.createShop)
+router.post(
+  "/create-product",
+  userAuthorization(["seller"]),
+  upload.array("images"),
+  transformFilesToBody,
+  validation(productSchema),
+  isProductExist,
+  productController.createProduct
+);
+router.post(
+  "/create-shop",
+  userAuthorization(["seller"]),
+  validation(shopSchema),
+  isShopExist,
+  productController.createShop
+);
+router.put(
+  "/update-product/:id",
+  userAuthorization(["seller"]),
+  upload.array("images"),
+  transformFilesToBody,
+  validation(productSchema),
+  isProductNotExist,

```

Comment:
Why created the **`isProductNotExist`** function while already having the **`isProductExist`** function ? just check and use it accordingly.

Reply:
isProductExist middleware checks if the product does already exist, and if it does it suggests the user to update the product instead.
But for updating product, we need to check if the product does not exist so that the seller can not attempt to update the product which does not exist, that's why I created isProductNotExist middleware.